boards.txt

```diff
@@ -2958,6 +2958,162 @@ S_ODI_Ultra.menu.EraseFlash.all.upload.erase_cmd=-e
+lilygo_t_display.menu.PartitionScheme.huge_app.upload.maximum_size=3145728
+lilygo_t_display.menu.PartitionScheme.min_spiffs=Minimal SPIFFS (1.9MB APP with OTA/190KB SPIFFS)
+lilygo_t_display.menu.PartitionScheme.min_spiffs.build.partitions=min_spiffs
+lilygo_t_display.menu.PartitionScheme.min_spiffs.upload.maximum_size=1966080
+lilygo_t_display.menu.PartitionScheme.fatflash=16M Flash (2MB APP/12.5MB FAT)
+lilygo_t_display.menu.PartitionScheme.fatflash.build.partitions=ffat
+lilygo_t_display.menu.PartitionScheme.fatflash.upload.maximum_size=2097152
+lilygo_t_display.menu.PartitionScheme.app3M_fat9M_16MB=16M Flash (3MB APP/9MB FATFS)
+lilygo_t_display.menu.PartitionScheme.app3M_fat9M_16MB.build.partitions=app3M_fat9M_16MB
+lilygo_t_display.menu.PartitionScheme.app3M_fat9M_16MB.upload.maximum_size=3145728
+
+lilygo_t_display.menu.CPUFreq.240=240MHz (WiFi/BT)
+lilygo_t_display.menu.CPUFreq.240.build.f_cpu=240000000L
+lilygo_t_display.menu.CPUFreq.160=160MHz (WiFi/BT)
+lilygo_t_display.menu.CPUFreq.160.build.f_cpu=160000000L
+lilygo_t_display.menu.CPUFreq.80=80MHz (WiFi/BT)
+lilygo_t_display.menu.CPUFreq.80.build.f_cpu=80000000L
+lilygo_t_display.menu.CPUFreq.40=40MHz (40MHz XTAL)
+lilygo_t_display.menu.CPUFreq.40.build.f_cpu=40000000L
+lilygo_t_display.menu.CPUFreq.26=26MHz (26MHz XTAL)
+lilygo_t_display.menu.CPUFreq.26.build.f_cpu=26000000L
+lilygo_t_display.menu.CPUFreq.20=20MHz (40MHz XTAL)
+lilygo_t_display.menu.CPUFreq.20.build.f_cpu=20000000L
+lilygo_t_display.menu.CPUFreq.13=13MHz (26MHz XTAL)
+lilygo_t_display.menu.CPUFreq.13.build.f_cpu=13000000L
+lilygo_t_display.menu.CPUFreq.10=10MHz (40MHz XTAL)
+lilygo_t_display.menu.CPUFreq.10.build.f_cpu=10000000L
+
+lilygo_t_display.menu.FlashMode.qio=QIO
+lilygo_t_display.menu.FlashMode.qio.build.flash_mode=dio
+lilygo_t_display.menu.FlashMode.qio.build.boot=qio
+lilygo_t_display.menu.FlashMode.dio=DIO
+lilygo_t_display.menu.FlashMode.dio.build.flash_mode=dio
+lilygo_t_display.menu.FlashMode.dio.build.boot=dio
+lilygo_t_display.menu.FlashMode.qout=QOUT
+lilygo_t_display.menu.FlashMode.qout.build.flash_mode=dout
+lilygo_t_display.menu.FlashMode.qout.build.boot=qout
+lilygo_t_display.menu.FlashMode.dout=DOUT
+lilygo_t_display.menu.FlashMode.dout.build.flash_mode=dout
+lilygo_t_display.menu.FlashMode.dout.build.boot=dout
+
+lilygo_t_display.menu.FlashFreq.80=80MHz
+lilygo_t_display.menu.FlashFreq.80.build.flash_freq=80m
+lilygo_t_display.menu.FlashFreq.40=40MHz
+lilygo_t_display.menu.FlashFreq.40.build.flash_freq=40m
+
+lilygo_t_display.menu.FlashSize.4M=4MB (32Mb)
+lilygo_t_display.menu.FlashSize.4M.build.flash_size=4MB
+lilygo_t_display.menu.FlashSize.8M=8MB (64Mb)
+lilygo_t_display.menu.FlashSize.8M.build.flash_size=8MB
+lilygo_t_display.menu.FlashSize.8M.build.partitions=default_8MB
+lilygo_t_display.menu.FlashSize.2M=2MB (16Mb)
+lilygo_t_display.menu.FlashSize.2M.build.flash_size=2MB
+lilygo_t_display.menu.FlashSize.2M.build.partitions=minimal
```

Comment:
As the boards will be 4MB or 16MB, this part can be removed and users will not be able to select wrong flash size.